This is a real failure in `QueueSinkSpec`, something is wrong with this backport, or we did something more in master.
It's supposed to be manual backport of #19783 and b4f507c

```
[info] - must work with one element buffer *** FAILED *** (6 seconds, 37 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsg while waiting for Some(2)
[info]   at scala.Predef$.assert(Predef.scala:179)
[info]   at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:338)
[info]   at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
[info]   at akka.stream.scaladsl.QueueSinkSpec$$anonfun$1$$anonfun$apply$mcV$sp$9$$anonfun$apply$mcV$sp$10.apply$mcV$sp(QueueSinkSpec.scala:113)
```

--- a/akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSinkSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSinkSpec.scala
@@ -301,6 +301,19 @@ class InputStreamSinkSpec extends AkkaSpec(UnboundedMailboxConfig) {
 
       inputStream.close()
     }
+
+    "fail to materialize with zero sized input buffer" in {
+      an[IllegalArgumentException] shouldBe thrownBy {
+        Source.single(byteString)
+          .runWith(StreamConverters.asInputStream(timeout).withAttributes(Attributes.inputBuffer(0, 0)))
+        /*
+       With Source.single we test the code path in which the sink
+       itself throws an exception when being materialized. If
+       Source.empty is used, the same exception is thrown by
+       Materializer.
+       */
+      }
+    }
   }
 
 }

--- a/akka-stream-tests/src/test/scala/akka/stream/io/OutputStreamSourceSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/io/OutputStreamSourceSpec.scala
@@ -11,7 +11,7 @@ import akka.stream._
 import akka.stream.impl.StreamSupervisor.Children
 import akka.stream.impl.io.OutputStreamSourceStage
 import akka.stream.impl.{ ActorMaterializerImpl, StreamSupervisor }
-import akka.stream.scaladsl.{ Keep, Source, StreamConverters }
+import akka.stream.scaladsl.{ Keep, Sink, Source, StreamConverters }
 import akka.stream.stage.OutHandler
 import akka.stream.testkit.Utils._
 import akka.stream.testkit._
@@ -174,6 +174,20 @@ class OutputStreamSourceSpec extends AkkaSpec(UnboundedMailboxConfig) {
       s.cancel()
       sourceProbe.expectMsg(OutputStreamSourceTestMessages.Finish)
       the[Exception] thrownBy outputStream.write(bytesArray) shouldBe a[IOException]
+    }
+
+    "fail to materialize with zero sized input buffer" in {
+      an[IllegalArgumentException] shouldBe thrownBy {
+        StreamConverters.asOutputStream(timeout)
+          .withAttributes(Attributes.inputBuffer(0, 0))
+          .runWith(Sink.head)
+        /*
+          With Sink.head we test the code path in which the source
+          itself throws an exception when being materialized. If
+          Sink.ignore is used, the same exception is thrown by
+          Materializer.
+         */
+      }
     }
   }
 }

--- a/akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefBackpressureSinkSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefBackpressureSinkSpec.scala
@@ -10,6 +10,7 @@ import akka.stream.testkit._
 import akka.stream.testkit.scaladsl._
 import org.scalatest.concurrent.ScalaFutures
 import org.scalactic.ConversionCheckedTripleEquals
+
 import scala.concurrent.duration._
 
 object ActorRefBackpressureSinkSpec {
@@ -160,6 +161,16 @@ class ActorRefBackpressureSinkSpec extends AkkaSpec with ScalaFutures with Conve
 
       publisher.sendComplete()
       expectMsg(completeMessage)
+    }
+
+    "fail to materialize with zero sized input buffer" in {
+      val fw = createActor(classOf[Fw])
+      an[IllegalArgumentException] shouldBe thrownBy {
+        val badSink = Sink
+          .actorRefWithAck(fw, initMessage, ackMessage, completeMessage)
+          .withAttributes(Attributes.inputBuffer(0, 0))
+        Source.single(()).runWith(badSink)
+      }
     }
 
   }

--- a/akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala
+++ b/akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala
@@ -116,5 +116,11 @@ class QueueSinkSpec extends AkkaSpec {
       Await.result(queue.pull(), noMsgTimeout) should be(None)
     }
 
+    "fail to materialize with zero sized input buffer" in {
+      an[IllegalArgumentException] shouldBe thrownBy {
+        Source.single(()).runWith(Sink.queue().withAttributes(Attributes.inputBuffer(0, 0)))
+      }
+    }
+
   }
 }

--- a/akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkStage.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/ActorRefBackpressureSinkStage.scala
@@ -8,10 +8,14 @@ import java.util
 import akka.actor._
 import akka.dispatch.sysmsg.{ DeathWatchNotification, SystemMessage, Watch }
 import akka.stream.stage.GraphStageLogic.StageActorRef
-import akka.stream.{ Inlet, SinkShape, ActorMaterializer, Attributes }
+import akka.stream.{ ActorMaterializer, Attributes, Inlet, SinkShape }
 import akka.stream.Attributes.InputBuffer
+import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.stage._
 
+/**
+ * INTERNAL API
+ */
 /**
  * INTERNAL API
  */
@@ -21,14 +25,15 @@ private[akka] class ActorRefBackpressureSinkStage[In](ref: ActorRef, onInitMessa
                                                       onFailureMessage: (Throwable) ⇒ Any)
   extends GraphStage[SinkShape[In]] {
   val in: Inlet[In] = Inlet[In]("ActorRefBackpressureSink.in")
+  override def initialAttributes = DefaultAttributes.actorRefWithAck
   override val shape: SinkShape[In] = SinkShape(in)
-
-  val maxBuffer = module.attributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
-  require(maxBuffer > 0, "Buffer size must be greater than 0")
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new GraphStageLogic(shape) {
       implicit var self: StageActorRef = _
+
+      val maxBuffer = inheritedAttributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
+      require(maxBuffer > 0, "Buffer size must be greater than 0")
 
       val buffer: util.Deque[In] = new util.ArrayDeque[In]()
       var acknowledgementReceived = false
@@ -36,9 +41,15 @@ private[akka] class ActorRefBackpressureSinkStage[In](ref: ActorRef, onInitMessa
 
       private def receive(evt: (ActorRef, Any)): Unit = {
         evt._2 match {
-          case `ackMessage` ⇒
-            if (!buffer.isEmpty) sendData()
-            else acknowledgementReceived = true
+          case `ackMessage` ⇒ {
+            if (buffer.isEmpty) acknowledgementReceived = true
+            else {
+              // onPush might have filled the buffer up and
+              // stopped pulling, so we pull here
+              if (buffer.size() == maxBuffer) tryPull(in)
+              dequeueAndSend()
+            }
+          }
           case Terminated(`ref`) ⇒ completeStage()
           case _                 ⇒ //ignore all other messages
         }
@@ -52,10 +63,8 @@ private[akka] class ActorRefBackpressureSinkStage[In](ref: ActorRef, onInitMessa
         pull(in)
       }
 
-      private def sendData(): Unit = {
-        if (buffer.size() == maxBuffer) tryPull(in)
+      private def dequeueAndSend(): Unit = {
         ref ! buffer.poll()
-        acknowledgementReceived = false
         if (buffer.isEmpty && completeReceived) finish()
       }
 
@@ -67,7 +76,10 @@ private[akka] class ActorRefBackpressureSinkStage[In](ref: ActorRef, onInitMessa
       setHandler(in, new InHandler {
         override def onPush(): Unit = {
           buffer offer grab(in)
-          if (acknowledgementReceived) sendData()
+          if (acknowledgementReceived) {
+            dequeueAndSend()
+            acknowledgementReceived = false
+          }
           if (buffer.size() < maxBuffer) pull(in)
         }
         override def onUpstreamFinish(): Unit = {

--- a/akka-stream/src/main/scala/akka/stream/impl/Sinks.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/Sinks.scala
@@ -3,12 +3,12 @@
  */
 package akka.stream.impl
 
-import java.util.concurrent.atomic.AtomicReference
-
 import akka.actor.{ ActorRef, Props }
 import akka.stream.Attributes.InputBuffer
 import akka.stream._
+import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.impl.StreamLayout.Module
+import akka.stream.impl.fusing.CallbackWrapper
 import akka.stream.stage.{ AsyncCallback, GraphStageLogic, GraphStageWithMaterializedValue, InHandler }
 import org.reactivestreams.{ Publisher, Subscriber }
 
@@ -241,33 +241,39 @@ private[akka] final class HeadOptionStage[T] extends GraphStageWithMaterializedV
 /**
  * INTERNAL API
  */
-private[akka] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkShape[T], SinkQueue[T]] {
-  trait RequestElementCallback[E] {
-    val requestElement = new AtomicReference[AnyRef](Nil)
-  }
-
-  type Requested[E] = Promise[Option[T]]
+/**
+ * INTERNAL API
+ */
+private[stream] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkShape[T], SinkQueue[T]] {
+  type Requested[E] = Promise[Option[E]]
 
   val in = Inlet[T]("queueSink.in")
+  override def initialAttributes = DefaultAttributes.queueSink
   override val shape: SinkShape[T] = SinkShape.of(in)
 
+  override def toString: String = "QueueSink"
+
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) = {
-    type Received[E] = Try[Option[E]]
+    val stageLogic = new GraphStageLogic(shape) with CallbackWrapper[Requested[T]] {
+      type Received[E] = Try[Option[E]]
 
-    val maxBuffer = module.attributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
-    require(maxBuffer > 0, "Buffer size must be greater than 0")
+      val maxBuffer = inheritedAttributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
+      require(maxBuffer > 0, "Buffer size must be greater than 0")
 
-    val buffer = FixedSizeBuffer[Received[T]](maxBuffer + 1)
-    var currentRequest: Option[Requested[T]] = None
-
-    val stageLogic = new GraphStageLogic(shape) with RequestElementCallback[Requested[T]] {
+      var buffer: Buffer[Received[T]] = _
+      var currentRequest: Option[Requested[T]] = None
 
       override def preStart(): Unit = {
+        // Allocates one additional element to hold stream
+        // closed/failure indicators
+        buffer = FixedSizeBuffer(maxBuffer + 1)
         setKeepGoing(true)
-        val list = requestElement.getAndSet(callback.invoke _).asInstanceOf[List[Requested[T]]]
-        list.reverse.foreach(callback.invoke)
+        initCallback(callback.invoke)
         pull(in)
       }
+
+      override def postStop(): Unit = stopCallback(promise ⇒
+        promise.failure(new IllegalStateException("Stream is terminated. QueueSink is detached")))
 
       private val callback: AsyncCallback[Requested[T]] =
         getAsyncCallback(promise ⇒ currentRequest match {
@@ -275,7 +281,10 @@ private[akka] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkS
             promise.failure(new IllegalStateException("You have to wait for previous future to be resolved to send another request"))
           case None ⇒
             if (buffer.isEmpty) currentRequest = Some(promise)
-            else sendDownstream(promise)
+            else {
+              if (buffer.used == maxBuffer) tryPull(in)
+              sendDownstream(promise)
+            }
         })
 
       def sendDownstream(promise: Requested[T]): Unit = {
@@ -301,7 +310,7 @@ private[akka] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkS
       setHandler(in, new InHandler {
         override def onPush(): Unit = {
           enqueueAndNotify(Success(Some(grab(in))))
-          if (buffer.used < maxBuffer - 1) pull(in)
+          if (buffer.used < maxBuffer) pull(in)
         }
         override def onUpstreamFinish(): Unit = enqueueAndNotify(Success(None))
         override def onUpstreamFailure(ex: Throwable): Unit = enqueueAndNotify(Failure(ex))
@@ -310,17 +319,8 @@ private[akka] class QueueSink[T]() extends GraphStageWithMaterializedValue[SinkS
 
     (stageLogic, new SinkQueue[T] {
       override def pull(): Future[Option[T]] = {
-        val ref = stageLogic.requestElement
         val p = Promise[Option[T]]
-        ref.get() match {
-          case l: List[_] ⇒
-            if (!ref.compareAndSet(l, p :: l))
-              ref.get() match {
-                case _: List[_]         ⇒ throw new IllegalStateException("Concurrent call of SinkQueue.pull() is detected")
-                case f: Function1[_, _] ⇒ f.asInstanceOf[Requested[T] ⇒ Unit](p)
-              }
-          case f: Function1[_, _] ⇒ f.asInstanceOf[Requested[T] ⇒ Unit](p)
-        }
+        stageLogic.invoke(p)
         p.future
       }
     })

--- a/akka-stream/src/main/scala/akka/stream/impl/Stages.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/Stages.scala
@@ -95,6 +95,7 @@ private[stream] object Stages {
     val fanoutPublisherSink = name("fanoutPublisherSink")
     val ignoreSink = name("ignoreSink")
     val actorRefSink = name("actorRefSink")
+    val actorRefWithAck = name("actorRefWithAckSink")
     val actorSubscriberSink = name("actorSubscriberSink")
     val queueSink = name("queueSink")
     val outputStreamSink = name("outputStreamSink") and IODispatcher

--- a/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala
@@ -3,12 +3,15 @@
  */
 package akka.stream.impl.fusing
 
-import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicReference }
+import java.util.concurrent.locks.ReentrantLock
+
 import akka.actor.Cancellable
 import akka.dispatch.ExecutionContexts
 import akka.event.Logging
 import akka.stream._
 import akka.stream.stage._
+
 import scala.concurrent.{ Future, Promise }
 import scala.concurrent.duration.FiniteDuration
 import akka.stream.impl.StreamLayout._
@@ -279,5 +282,57 @@ object GraphStages {
       })
     }
     override def toString: String = s"SingleSource($elem)"
+  }
+}
+
+/**
+ * INTERNAL API
+ * This trait wraps callback for `GraphStage` stage instances and handle gracefully cases when stage is
+ * not yet initialized or already finished.
+ *
+ * While `GraphStage` has not initialized it adds all requests to list.
+ * As soon as `GraphStage` is started it stops collecting requests (pointing to real callback
+ * function) and run all the callbacks from the list
+ *
+ * Supposed to be used by GraphStages that share call back to outer world
+ */
+private[akka] trait CallbackWrapper[T] extends AsyncCallback[T] {
+  private trait CallbackState
+  private case class NotInitialized(list: List[T]) extends CallbackState
+  private case class Initialized(f: T ⇒ Unit) extends CallbackState
+  private case class Stopped(f: T ⇒ Unit) extends CallbackState
+
+  /*
+   * To preserve message order when switching between not initialized / initialized states
+   * lock is used. Case is similar to RepointableActorRef
+   */
+  private[this] final val lock = new ReentrantLock
+
+  private[this] val callbackState = new AtomicReference[CallbackState](NotInitialized(Nil))
+
+  def stopCallback(f: T ⇒ Unit): Unit = locked {
+    callbackState.set(Stopped(f))
+  }
+
+  def initCallback(f: T ⇒ Unit): Unit = locked {
+    val list = (callbackState.getAndSet(Initialized(f)): @unchecked) match {
+      case NotInitialized(l) ⇒ l
+    }
+    list.reverse.foreach(f)
+  }
+
+  override def invoke(arg: T): Unit = locked {
+    callbackState.get() match {
+      case Initialized(cb)          ⇒ cb(arg)
+      case list @ NotInitialized(l) ⇒ callbackState.compareAndSet(list, NotInitialized(arg :: l))
+      case Stopped(cb) ⇒
+        lock.unlock()
+        cb(arg)
+    }
+  }
+
+  private[this] def locked(body: ⇒ Unit): Unit = {
+    lock.lock()
+    try body finally if (lock.isLocked) lock.unlock()
   }
 }

--- a/akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala
@@ -5,13 +5,16 @@ package akka.stream.impl.io
 
 import java.io.{ IOException, InputStream }
 import java.util.concurrent.{ BlockingQueue, LinkedBlockingDeque, TimeUnit }
+
 import akka.stream.Attributes.InputBuffer
+import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.impl.io.InputStreamSinkStage._
 import akka.stream.stage._
 import akka.util.ByteString
+
 import scala.annotation.tailrec
 import scala.concurrent.duration.FiniteDuration
-import akka.stream.{ Inlet, SinkShape, Attributes }
+import akka.stream.{ Attributes, Inlet, SinkShape }
 
 private[akka] object InputStreamSinkStage {
 
@@ -37,12 +40,12 @@ private[akka] class InputStreamSinkStage(readTimeout: FiniteDuration) extends Gr
 
   val in = Inlet[ByteString]("InputStreamSink.in")
   override val shape: SinkShape[ByteString] = SinkShape.of(in)
-
-  // has to be in this order as module depends on shape
-  val maxBuffer = module.attributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
-  require(maxBuffer > 0, "Buffer size must be greater than 0")
+  override def initialAttributes: Attributes = DefaultAttributes.inputStreamSink
 
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, InputStream) = {
+    val maxBuffer = inheritedAttributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
+    require(maxBuffer > 0, "Buffer size must be greater than 0")
+
     val dataQueue = new LinkedBlockingDeque[StreamToAdapterMessage](maxBuffer + 2)
 
     val logic = new GraphStageLogic(shape) with StageWithCallback {

--- a/akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala
@@ -7,8 +7,9 @@ import java.io.{ IOException, OutputStream }
 import java.util.concurrent.atomic.AtomicReference
 import java.util.concurrent.{ BlockingQueue, LinkedBlockingQueue }
 
-import akka.stream.{ Outlet, SourceShape, Attributes }
+import akka.stream.{ Attributes, Outlet, SourceShape }
 import akka.stream.Attributes.InputBuffer
+import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.impl.io.OutputStreamSourceStage._
 import akka.stream.stage._
 import akka.util.ByteString
@@ -35,12 +36,13 @@ private[akka] object OutputStreamSourceStage {
 private[akka] class OutputStreamSourceStage(writeTimeout: FiniteDuration) extends GraphStageWithMaterializedValue[SourceShape[ByteString], OutputStream] {
   val out = Outlet[ByteString]("OutputStreamSource.out")
   override val shape: SourceShape[ByteString] = SourceShape.of(out)
-
-  // has to be in this order as module depends on shape
-  val maxBuffer = module.attributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
-  require(maxBuffer > 0, "Buffer size must be greater than 0")
+  override def initialAttributes = DefaultAttributes.outputStreamSource
 
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, OutputStream) = {
+    // has to be in this order as module depends on shape
+    val maxBuffer = inheritedAttributes.getAttribute(classOf[InputBuffer], InputBuffer(16, 16)).max
+    require(maxBuffer > 0, "Buffer size must be greater than 0")
+
     val dataQueue = new LinkedBlockingQueue[ByteString](maxBuffer)
     val downstreamStatus = new AtomicReference[DownstreamStatus](Ok)
 

--- a/akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala
+++ b/akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala
@@ -886,7 +886,7 @@ object GraphDSL extends GraphApply {
     }
 
     private class PortOpsImpl[+Out](override val outlet: Outlet[Out @uncheckedVariance], b: Builder[_])
-        extends PortOps[Out] {
+      extends PortOps[Out] {
 
       override def withAttributes(attr: Attributes): Repr[Out] =
         throw new UnsupportedOperationException("Cannot set attributes on chained ops from a junction output port")

--- a/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala
+++ b/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala
@@ -318,6 +318,6 @@ object Sink {
    * @see [[akka.stream.SinkQueue]]
    */
   def queue[T](): Sink[T, SinkQueue[T]] =
-    Sink.fromGraph(new QueueSink().withAttributes(DefaultAttributes.queueSink))
+    Sink.fromGraph(new QueueSink())
 
 }

--- a/akka-stream/src/main/scala/akka/stream/scaladsl/StreamConverters.scala
+++ b/akka-stream/src/main/scala/akka/stream/scaladsl/StreamConverters.scala
@@ -49,7 +49,7 @@ object StreamConverters {
    * @param writeTimeout the max time the write operation on the materialized OutputStream should block, defaults to 5 seconds
    */
   def asOutputStream(writeTimeout: FiniteDuration = 5.seconds): Source[ByteString, OutputStream] =
-    Source.fromGraph(new OutputStreamSourceStage(writeTimeout)).withAttributes(DefaultAttributes.outputStreamSource)
+    Source.fromGraph(new OutputStreamSourceStage(writeTimeout))
 
   /**
    * Creates a Sink which writes incoming [[ByteString]]s to an [[OutputStream]] created by the given function.
@@ -87,6 +87,6 @@ object StreamConverters {
    * @param readTimeout the max time the read operation on the materialized InputStream should block
    */
   def asInputStream(readTimeout: FiniteDuration = 5.seconds): Sink[ByteString, InputStream] =
-    Sink.fromGraph(new InputStreamSinkStage(readTimeout)).withAttributes(DefaultAttributes.inputStreamSink)
+    Sink.fromGraph(new InputStreamSinkStage(readTimeout))
 
 }

--- a/project/AkkaBuild.scala
+++ b/project/AkkaBuild.scala
@@ -1336,13 +1336,22 @@ object AkkaBuild extends Build {
       // Removed internal methods https://github.com/akka/akka/pull/20162/files
       ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.fusing.GraphInterpreter.fail"),
       ProblemFilters.exclude[MissingMethodProblem]("akka.stream.stage.GraphStageLogic.failStage"),
-      
+
       ProblemFilters.exclude[FinalClassProblem]("akka.stream.impl.fusing.GraphStages$Breaker"),
 
       // #19589 auto flushing (params on internal classes changed)
       ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.OutputStreamSubscriber.this"),
       ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.OutputStreamSubscriber.props"),
-      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.OutputStreamSink.this")
+      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.OutputStreamSink.this"),
+      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.stage.GraphStageLogic.failStage"),
+
+      // #19783
+      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.ActorRefBackpressureSinkStage.maxBuffer"),
+      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.OutputStreamSourceStage.maxBuffer"),
+      ProblemFilters.exclude[MissingMethodProblem]("akka.stream.impl.io.InputStreamSinkStage.maxBuffer"),
+
+      // #Changing QuueueSink internal implementation
+      ProblemFilters.exclude[MissingClassProblem]("akka.stream.impl.QueueSink$RequestElementCallback")
     )
   }
 

